src/vs/platform/quickinput/browser/quickInput.ts

```diff
@@ -227,4 +230,75 @@ export class QuickInputService extends Themable implements IQuickInputService {
 			})
 		};
 	}
+
+	formatPinnedItems(storageKey: string, quickPick: IQuickPick<IQuickPickItem>, storageService: IStorageService, callback?: () => Promise<void>): void {
+		quickPick.onDidTriggerItemButton((e) => {
+			if (e.button.iconClass === ThemeIcon.asClassName(Codicon.pin) || e.button.iconClass !== ThemeIcon.asClassName(Codicon.pinned)) {
+				this._formatPinnedItems(storageKey, quickPick, storageService, callback, e);
```

Comment:
Is callback used? Instead of passing `e` in here, I was expecting the button to update state in storage and then calling `callback()` to reopen the quick pick.

Reply:
see https://github.com/microsoft/vscode/pull/158037/files#r944776771 - not currently. I like your idea above better